### PR TITLE
Downgrade matplotlib dependency to 3.2 and fix numerous pylint issues

### PR DESCRIPTION
**Pull request by canyon289**
Testing to see what happens if we lower the matplotlib dependency. This is for google colab update

----
:books: Documentation preview :books:: https://arviz--2201.org.readthedocs.build/en/2201/

### CODE REVIEW

**Comment by canyon289** [comment]
ArviZ works on a lower version of matplotlib. To get google colab updated for both arviz and pymc I suggest we lower the minimum version to 3.2. Any issues if we do this and cut a release?

**Comment by canyon289** [comment]
Anyone know why we get pylint errors with no code changes?

**Comment by OriolAbril** [comment]
New pylint version which is more strict I think

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 setuptools>=60.0.0
-matplotlib>=3.5
+matplotlib~=3.2
```
Comment (OriolAbril) [inline requirements.txt:2]: Remember to update this to >

**Review by OriolAbril — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (canyon289) [reply requirements.txt]: Thanks

**Comment by canyon289** [comment]
This is ready for rereview

**Review by ahartikainen — APPROVED** [review]

**Pull request merged**